docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+```json
+{
+        "contextDefinitions": [{
+                "name": "Patient",
+                "primaryDataType": "Patient",
+                "primaryKeyColumn": "patient_id",
+				"relationships":[{
+					"type": "OneToMany",
+					"relatedDataType": "Observation",
+					"relatedKeyColumn": "patient_id"
+				},{
+					"type": "OneToMany",
+					"relatedDataType": "Condition",
+					"relatedKeyColumn": "patient_id"
+				}]
+        },{
+                "name": "Claim",
+                "primaryDataType": "Claim",
+                "primaryKeyColumn": "claim_id",
+				"relationships": [{
+					"type": "ManyToMany",
+					"associationDataType": "PatientClaim",
+					"associationOneKeyColumn": "claim_id",
+					"associationManyKeyColumn": "patient_id",
+					"relatedDataType": "Patient",
+					"relatedKeyColumn": "patient_id"
+				}]
+        }]
+}
+```
+
+For the initial pilot implementation of the Spark application, join logic is limited to the one-to-many and many-to-many scenarios described above. More complex join logic involving compound primary keys (more than one column required to uniquely identify a record) or related data beyond a single table relationship (with or without an association table in between) are unsupported. These will come in a future release.
+
+If the user wishes to limit which aggregation contexts are evaluated during an application run, there is a program option (`-a`) that can be specified one more more times that lists the contextKey value for each aggregation context that should be evaluated.
+
+### CQL Job Descriptions
+
+The CQL jobs that will be evaluated by a run of the Spark CQL Evaluator application are stored in a JSON-formatted configuration file that is typically called cql-jobs.json. Again, the name isn't prescribed, but the rest of this guide will use the name cql-jobs.json for simplicity. The cql-jobs.json file lists each CQL Library that will be used, which expressions (aka define statements) within the CQL Library will be evaluated, and any input parameters that will be passed to the CQL evaluation engine to be used by parameterized logic within the library, and a `contextKey` that links the CQL to a particular aggregation context defined in the `context-definitions.json` file. A convenience item is provided for configuring CQL parameters at a global level. This is potentially useful for something like a measurement period that is intended to be constant throughout the evaluation of all libraries and statements being evaluated.
```

Comment:
Maybe there's a good reason we did it this way - or maybe this is something we could consider as a future enhancement, but given a CQL library and a define statement, could we not determine the context that is in effect for that define statement, versus having it explicitly called out in cql-jobs?

Reply:
A CQL library can technically have 0..n `context` statements (e.g. CQF Measures IG [here](https://hl7.org/fhir/us/cqfmeasures/2021May/measure-conformance.html#patient-based-calculation-1)). Even if we assumed that each library had exactly one `context` statement, I like that the current design is flexible enough to all the user to execute the same library across different *aggregation* contexts. For example, one job might join patient and eligibility and another job might join patient and claims data. Both are "context Patient" from the perspective of the CQL, but the underlying data we collect is different. I don't know how much people would actually use this, but it seems plausible to split up jobs this way. I'd be willing to revisit if there is a strong feeling that we should limit the CQL to exactly one aggregation context.